src/validation/validationSchema.ts

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+export default {
+    Common: {
+        Next: true,
+        End: true,
+        Comment: true,
+        InputPath: true,
+        OutputPath: true,
+        Type: true
+    },
+    Specific: {
+        Pass: {
+            Properties: {
+                Result: true,
+                ResultPath: true,
+                Parameters: true
+            }
+        },
+        Task: {
+            Properties: {
+                Resource: true,
+                Parameters: true,
+                ResultPath: true,
+                TimeoutSeconds: true,
+                HeartbeatSeconds: true,
+                Retry: {
+                    'Fn:ArrayOf': 'Retry'
+                },
+                Catch: {
+                    'Fn:ArrayOf': 'Catch'
+                }
+            }
+        },
+        Choice: {
+            Properties: {
+                Comment: true,
+                InputPath: true,
+                OutputPath: true,
+                Type: true,
+                Choices: {
+                    'Fn:ArrayOf': 'Choices'
+                },
+                Default: true,
+                hasCommonFields: false
+            }
```

Comment:
Yes, `hasCommonFields` was in the wrong place. I inverted it as well so that it has to be specified as `true` for grater clarity. Your other suggestions are very valid and I was considering it as well but I think it would be an overkill for now. Definitely something to consider when we actually want to implement our own schema parser to replace JSON Schema.